src/compiler/program.ts

```diff
@@ -1076,11 +1076,6 @@ namespace ts {
                 !options.experimentalDecorators) {
                 programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "emitDecoratorMetadata", "experimentalDecorators"));
             }
-
-            if (options.experimentalAsyncFunctions &&
-                options.target !== ScriptTarget.ES6) {

```

Comment:
do we now have ES5 support for async functions? What happens if you target ES5 and then use an async function?